crates/rattler_repodata_gateway/src/gateway/query.rs

```diff
@@ -117,46 +144,75 @@ impl GatewayQuery {
             });
         }
 
-        // Package names that we have or will issue requests for.
-        let mut seen = HashSet::new();
-        let mut pending_package_specs = HashMap::new();
-        for spec in self.specs {
-            if let Some(name) = &spec.name {
-                seen.insert(name.clone());
-                pending_package_specs
-                    .entry(name.clone())
-                    .or_insert_with(Vec::new)
-                    .push(spec);
-            }
-        }
-
-        // A list of futures to fetch the records for the pending package names. The main task
-        // awaits these futures.
+        // A list of futures to fetch the records for the pending package names.
+        // The main task awaits these futures.
         let mut pending_records = FuturesUnordered::new();
 
-        // The resulting list of repodata records.
-        let mut result = vec![RepoData::default(); subdirs.len()];
+        // Push the direct url queries to the pending_records.
+        for spec in direct_url_specs.clone() {
```

Comment:
```suggestion
        for (spec, url) in direct_url_specs {
```

Is the clone needed?

Reply:
Nope :+1: